logstreams/src/main/java/io/camunda/zeebe/logstreams/log/LogStreamWriter.java

```diff
@@ -22,14 +39,65 @@ public interface LogStreamWriter {
+  }
 
   /**
-   * Attempts to write the event to the underlying stream.
+   * Attempts to write the events to the underlying stream. This method is atomic, either all events
+   * are written, or none are.
    *
-   * @param appendEntry the entry to write
-   * @param sourcePosition a back-pointer to the record whose processing created this entry
-   * @return the event position, a negative value if fails to write the event, or 0 if the value is
-   *     empty
+   * @param appendEntries a set of entries to append; these will be appended in the order
+   * @return the last (i.e. highest) event position, a negative value if fails to write the events,
+   *     or 0 if the batch is empty
+   */
+  default long tryWrite(final LogAppendEntry... appendEntries) {
+    return tryWrite(Arrays.asList(appendEntries));
+  }
+
+  /**
+   * Attempts to write the events to the underlying stream. This method is atomic, either all events
+   * are written, or none are.
+   *
+   * @param appendEntries a set of entries to append; these will be appended in the order in which
+   *     the collection is iterated.
+   * @return the last (i.e. highest) event position, a negative value if fails to write the events,
+   *     or 0 if the batch is empty
+   */
+  default long tryWrite(final Iterable<? extends LogAppendEntry> appendEntries) {
+    return tryWrite(appendEntries, LogEntryDescriptor.KEY_NULL_VALUE);
+  }
+
+  /**
+   * Attempts to write the events to the underlying stream. This method is atomic, either all events
+   * are written, or none are.
+   *
+   * @param appendEntries a set of entries to append; these will be appended in the order in which
+   *     the collection is iterated.
+   * @param sourcePosition a back-pointer to the record whose processing created these entries
+   * @return the last (i.e. highest) event position, a negative value if fails to write the events,
+   *     or 0 if the batch is empty
+   */
+  default long tryWrite(
+      final Iterable<? extends LogAppendEntry> appendEntries, final long sourcePosition) {
+    return tryWrite(
+        StreamSupport.stream(
+                appendEntries::spliterator,
+                Spliterator.IMMUTABLE | Spliterator.NONNULL | Spliterator.ORDERED,
+                false)
+            .toList(),
+        sourcePosition);
+  }
+
+  /**
+   * Attempts to write the events to the underlying stream. This method is atomic, either all events
+   * are written, or none are.
+   *
+   * @param appendEntries a list of entries to append; append order is maintained
+   * @param sourcePosition a back-pointer to the record whose processing created these entries
+   * @return the last (i.e. highest) event position, a negative value if fails to write the events,
+   *     or 0 if the batch is empty
    */
-  long tryWrite(final LogAppendEntry appendEntry, final long sourcePosition);
+  long tryWrite(final List<? extends LogAppendEntry> appendEntries, final long sourcePosition);
```

Comment:
:question: Isn't our usage mostly just limited to `Collection`? Do we rely on the `List` contract?